### DISCUSSION TITLE
Server updates

### DISCUSSION THREAD

**Discussion by FordMustang0288** [Nimdy/Dedicated_Valheim_Server_Script#105] [Q&A]
Hello,
I see that a new dedicated server update is available which introduces a menu. As an existing user, do I need to do anything other than restart my server? Or are there specific upgrade instructions that I've missed?

Thanks

**Comment by FordMustang0288** [comment]
In terms of restarting, the main instructions for the original install provides the guidance you are looking for.
Stop server, restart...and the server starts automatically on reboot. You can view the status after a few moments.

**Comment by Nimdy** [comment]
never mind the update script works.

I just verified it on both my boxes.... not need to reboot

both the dev box and production box.

**Comment by Nimdy** [comment]
also you can just update the menu... everything will be fine

The only thing it does is add improvements
you will not lose anything
and it will make life better

**Comment by FordMustang0288** [comment]
@Nimdy - I have your "old" version installed so I do not have menu.sh.

If I try to clone to tmp and copy menu.sh onto my server, OPTION 1 says I would have a merge conflict with menu.sh if I tried to update.

So the question is, in /opt...if I clone to a folder called Valheim_Server_Latest. And then I choose INSTALL server. Will this overwrite my server that is running with your old method? I did back up my world files, but I am afraid to proceed. I would like some more clarification of how I can get my server up to date since I do not yet have the menu script.

**Comment by Nimdy** [comment]
@FordMustang0288 no issues updating the script.

Just remove it the old folder
cd /opt
rm -Rf Dedicated_Server_Valheim_Server
git clone https://github.com/Nimdy/Dedicated_Valheim_Server_Script.git

After that you will not have any issues.

We made the script to only manage your Valheim server not change all the files around, so if I die... your server breaks.  

From here on out... the menu should update just right... :)

**Comment by FordMustang0288** [comment]
Upon doing the clone, will I need to choose to install/reinstall the server. Or ignore that entirely and simply try to start the server?

Thank you!

**Comment by Nimdy** [comment]
You can if you want to but know your server start_valheim.sh configure information.

Because if you install/reinstall the server it will ask you for this information and rebuild everything again and save your information for later. However you will not lose any data and the new menu system has a added function to edit your configuration file as needed.

This information is now saved in two locations... within the /home/steam dir and the start_valheim.sh dir

This is for your history info of settings and start_valheim.sh requirements.

Your choice.  but you will not need to reinstall/install anything... just redownload the git clone and chmod +x menu.sh and then launch it with ./menu.sh
